rhesus-monkeys/scripts/convert-meta.py

```diff
@@ -0,0 +1,97 @@
+df_infant_w2 = df08.select(pl.col("Infant_ID"), pl.col("Infant_weight").alias("weight_at_x_days_old"), pl.col("PD").alias("x_days_old"))
+df_infant_weight = pl.concat([df_infant_w1, df_infant_w2])
+df_infant_weight1 = df_infant_weight.unique()
+df_infant_weight1.write_csv("../data-raw/data_infant_weight.csv", separator=";")
+
+# Create the adult metadata files
+    # FIO There are no additional Mother_ID in file 11
+df09 = pl.read_csv(folder_path / "data09.csv")
+df10 = pl.read_csv(folder_path / "data10.csv")
+df11 = pl.read_csv(folder_path / "data11.csv")
+
+df_adult09 = df09.select(pl.col("Mother_ID"), pl.col("Mother_age").alias("Age_at_conception"), pl.col("GD_Delivery").alias("gestation_day_at_delivery")
+, pl.col("Group").alias("obesity_classification"), pl.col("Mode_birth"), pl.col("Reject"), pl.col("Infant_sex"))
+df_adult10 = df10.select(pl.col("Mother_ID"), pl.col("Mother_age").alias("Age_at_conception"), pl.col("GD_Delivery").alias("gestation_day_at_delivery")
+, pl.col("Group").alias("obesity_classification"), pl.col("Mode_birth"), pl.col("Reject"), pl.col("Infant_sex"))
+
+df_adult_w1 = df09.select(pl.col("Mother_ID"), pl.col("GD_day").alias("sample_gestation_day"), pl.col("GD_targeted").alias("target_gestation_day")
+, pl.col("Mother_Weight").alias("weight_at_gestation_day"), pl.col("BCS").alias("body_condition_score")) 
+df_adult_w2 = df10.select(pl.col("Mother_ID"), pl.col("GD").alias("sample_gestation_day"), pl.col("Target_GD").alias("target_gestation_day")
+, pl.col("Mother_Weight").alias("weight_at_gestation_day"), pl.col("BCS").alias("body_condition_score")) 
+
+# Concatenate and de-dup adult files
+df_adult1 = pl.concat([df_adult09, df_adult10])
+df_adult2 = df_adult1.unique()
+df_adult2.write_csv("../data-raw/data_adult_meta.csv", separator=";")
+
+df_adultw1 = pl.concat([df_adult_w1, df_adult_w2])
+df_adultw2 = df_adultw1.unique()
+df_adultw2.write_csv("../data-raw/data_adult_weight.csv", separator=";")
+
+# Create the linking table
+
+df_link07 = df07.select(pl.col("Infant_ID"), pl.col("Exp"), pl.col("PD").alias("day_sample_taken"), pl.col("Batch"))
+df_link08 = df08.select(pl.col("Infant_ID"), pl.col("Exp"), pl.col("PD").alias("day_sample_taken"), pl.col("Batch"))
+
+df_link09 = df09.select(pl.col("Mother_ID"), pl.col("Exp"), pl.col("GD_day").alias("day_sample_taken"), pl.col("GD_targeted").alias("target_sampling_day")
+, pl.col("Batch"), pl.col("Dilution_factor"))
+df_link10 = df10.select(pl.col("Mother_ID"), pl.col("Exp"), pl.col("GD").alias("day_sample_taken"), pl.col("Target_GD").alias("target_sampling_day")
+, pl.col("Batch"), pl.col("Dilution_factor"))
+df_link11 = df11.select(pl.col("Mother_ID"), pl.col("Exp"), pl.col("GD").alias("day_sample_taken"), pl.col("Target_GD").alias("target_sampling_day")
+, pl.col("Batch"), pl.col("Dilution_factor"))
+
+matter1 = "Blood"
+matter2 = "Urine"
+matter3 = "Placenta"
+
+df_lnk1 = df_link07.with_columns(pl.lit(matter1).alias("type_of_matter"))
```

Comment:
I'm not sure what the `with_columns()` is needed or used for here. Isn't this a rename of a column?